kie-maven-plugin/src/main/java/org/kie/maven/plugin/BuildMojo.java

```diff
@@ -247,76 +275,21 @@ private void saveFile(MemoryFileSystem mfs, String fileName) throws MojoFailureE
-            if (kModule != null) {
-                optionalKieMap.get().put(sbkModule.toString(),
-                        kModule);
-                getLog().info("KieModule available in the map shared with the Maven Embedder");
-            }
-        }
-    }
-
-    private void shareStoreWithMap(ClassLoader classLoader) {
-        Optional<Map<String, Object>> optionalKieMap = getKieMap();
-        if (optionalKieMap.isPresent() && classLoader instanceof ProjectClassLoader) {
-            ProjectClassLoader projectClassloder = (ProjectClassLoader) classLoader;
-            Map<String, byte[]> types = projectClassloder.getStore();
-            if (projectClassloder.getStore() != null) {
-                StringBuilder sbTypes = new StringBuilder(compilationID).append(".").append("ProjectClassloaderStore");
-                optionalKieMap.get().put(sbTypes.toString(), types);
-                getLog().info("ProjectClassloader Store available in the map shared with the Maven Embedder");
-            }
-        }
-    }
-
-    private void shareTypesMetaInfoWithMap(InternalKieModule kModule) {
-        Optional<Map<String, Object>> optionalKieMap = getKieMap();
-        if (optionalKieMap.isPresent()) {
-            KieMetaInfoBuilder kb = new KieMetaInfoBuilder(kModule);
-            KieModuleMetaInfo info = kb.getKieModuleMetaInfo();
-            Map <String, TypeMetaInfo> typesMetaInfos =  info.getTypeMetaInfos();
-
-            if(typesMetaInfos != null){
-                StringBuilder sbTypes = new StringBuilder(compilationID).append(".").append(TypeMetaInfo.class.getName());
-                Set<String> eventClasses = new HashSet<>();
-                for(Map.Entry<String,TypeMetaInfo> item :typesMetaInfos.entrySet()) {
-                    if (item.getValue().isEvent()){
-                        eventClasses.add(item.getKey());
-                    }
-                }
-                optionalKieMap.get().put(sbTypes.toString(), eventClasses);
-                getLog().info("TypesMetaInfo keys available in the map shared with the Maven Embedder");
-            }
-        }
-    }
 
 
-    private Optional<Map<String, Object>> getKieMap() {
+    private Map<String, Object> getKieMap() {
         try {
             /**
              * Retrieve the map passed into the Plexus container by the MavenEmbedder from the MavenIncrementalCompiler in the kie-wb-common
              */
             Map<String, Object> kieMap = (Map) container.lookup(Map.class,
-                    "java.util.HashMap",
-                    "kieMap");
-            return Optional.of(kieMap);
+                                                                "java.util.HashMap",
+                                                                "kieMap");
+            return Optional.ofNullable(kieMap).orElse(Collections.emptyMap());
         } catch (ComponentLookupException cle) {
             getLog().info("kieMap not present with compilationID and container present");
```

Comment:
Shouldn't this be warning instead of info message?
The message "kieMap not present with compilationID and container present" is a bit confusing
Is the kie map not present? Or is it present but doesn't contain complationID? 
Imagine that you are a user who sees this log message. If the message is good he'll immediately know what's wrong and what and where should be fixed. Otherwise he'll have to look at the source code to understand what's wrong :(

Also there is probably duplication of log messages on line 194: `getLog().info("Kie Map not present");` - we should probably only log this at one place.

Reply:
I'll change with more clear messages